src/app/forms/validators/adr/adr.validators.ts

```diff
@@ -67,4 +69,43 @@ export class AdrValidators {
 
 		return null;
 	};
+
+	/**
+	 * Body declaration should only be visible when:
+	 *  - Carries dangerous goods is true
+	 *  - AND The ADR body type is one of the following:
+	 *    > Rigid box body
+	 *    > Full drawbar box body
+	 *    > Centre axle box body
+	 *    > Semi trailer box body
+	 *  - AND The permitted dangerous good 'Explosive (type 3)' is selected
+	 *
+	 * @param control
+	 */
+	static setBodyDeclarationVisibility = (): ValidatorFn => {
+		return (control: AbstractControl) => {
+			const carriesDangerousGoods = control.parent?.get('techRecord_adrDetails_dangerousGoods')?.value;
+			const permittedDangerousGoods = control.parent?.get('techRecord_adrDetails_permittedDangerousGoods')?.value;
+			const adrBodyType = control.parent?.get('techRecord_adrDetails_vehicleDetails_type')?.value;
```

Comment:
We could move these inside the `if` check, it seems they're only relevant when that check is satisfied

Reply:
Can do, suppose the form control name is already pretty descriptive so no need for the duplication.